### PR TITLE
Adding repl

### PR DESCRIPTION
**Pull request by donovanh**
Adds a simple REPL people can hack in - plus a mechanism for wrapping existing code in:

```
<div class="repl-code">
...
</div>
```

It'll then generate a link to the repl with the code pre-populated. Example:

repl 2019-11-11 21_21_30

### CODE REVIEW

**Review by readikus — APPROVED** [review]

**Comment by donovanh** [comment]
Wondering if I should adapt it to not automatically stick a link under every code block? Maybe that's too much. Opines plz

File: js/main.js [hunk 1]
```diff
@@ -1,8 +1,29 @@
+/* Add an edit in repl link to code blocks */
+
+function checkForReplLinks() {
+  const replCode = document.querySelectorAll("pre");
+  [...replCode].forEach(pre => {
+    console.log(pre);
```
Comment (donovanh) [inline js/main.js]: I need to remove this. Plus maybe revisit this wall-o-text

File: repl/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+---
+layout: default.njk
```
Comment (donovanh) [inline repl/index.md]: Need to remove the .njk

**Review by jjmax75 — APPROVED** [review]
LGTM

The last merge seems to have caused some conflicts with this branch but once it's all sorted I can merge this in

**Pull request merged**